provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/AbstractCrossDCTest.java

```diff
@@ -113,5 +120,27 @@ public void tearDownTestEnvironment() {
                 });
 
         MOCK_COOKIE_MANAGER.getCookieStore().removeAll();
+
+        DC_1.kc().markLBCheckUp();
+        DC_2.kc().markLBCheckUp();
+        waitUntilPrimaryIsUp(LOAD_BALANCER_KEYCLOAK.getKeycloakServerUrl(), DC_1.getKeycloakServerURL(), DC_2.getKeycloakServerURL());
+    }
+
+    public void waitForFailover(String clientUrl, String primaryUrl, String backupUrl) throws IOException, InterruptedException {
```

Comment:
How would this end if the failover doesnt happen for some reason ?

Reply:
Failover should finish when AWS health check detects that a site is down (refreshed every 30s) + when DNS records are updated (depends on DNS caching setup). It usually takes around 1.5 - 2 minutes.  
When failover fails for whatever reason the test will probably hang there. The easiest thing we can do is to add `@Timeout(value = 10, unit = TimeUnit.MINUTES)`  annotation on the test.

Reply:
I think thats a good idea either way, lets do that to add a timeout at the testcase level @martin-kanis